server/src/main/java/org/elasticsearch/action/admin/cluster/configuration/TransportClearVotingConfigExclusionsAction.java

```diff
@@ -105,7 +105,8 @@ public void onTimeout(TimeValue timeout) {
 
     private void submitClearVotingConfigExclusionsTask(ClearVotingConfigExclusionsRequest request, long startTimeMillis,
                                                        ActionListener<ClearVotingConfigExclusionsResponse> listener) {
-        clusterService.submitStateUpdateTask("clear-voting-config-exclusions", new ClusterStateUpdateTask(Priority.URGENT) {
+        clusterService.submitStateUpdateTask("clear-voting-config-exclusions", new ClusterStateUpdateTask(Priority.URGENT,
+                TimeValue.timeValueMillis(request.getTimeout().millis() + startTimeMillis - threadPool.relativeTimeInMillis())) {
```

Comment:
This seems unsafe in that it could be negative and cause an exception. Your change reduces the risk of that, but I wonder if we should fix it in this same PR?

Reply:
I pushed 5897ae9, that should give us a consistent (as in same as when the update would timeout on the queue) exception here I think.